protoc-gen-kotlin/lib/src/commonMain/kotlin/pbandk/gen/CodeGenerator.kt

```diff
@@ -125,31 +172,89 @@ open class CodeGenerator(
+                line("fun addFields${index}(fieldsList: ArrayList<pbandk.FieldDescriptor<$fullTypeName, *>>) {").indented {
+                    addFields(chunk, fullTypeName)
+                }.line("}")
+            }
+        }
+    }
+
+    private fun addFields(
+        chunk: List<Pair<File.Field.Numbered, File.Field.OneOf?>>,
+        fullTypeName: String
+    ) {
+        line("fieldsList.apply {").indented {
+            chunk.forEach { (field, oneof) ->
+                line("add(").indented {
+                    line("pbandk.FieldDescriptor(").indented {
+                        generateFieldDescriptorConstructorValues(
+                            field,
+                            fullTypeName,
+                            oneof,
+                            "this@Companion::descriptor"
+                        )
+                    }.line(")")
+                }.line(")")
+            }
+        }.line("}")
+    }
+
+    private fun generateFieldOptions(fieldOptions: FieldOptions) {
+        // We don't use/support other field option values currently. Once we support all of the options, this check
+        // should change to `fieldOptions != FieldOptions.defaultInstance`
+        if (fieldOptions.deprecated != null || fieldOptions.unknownFields.isNotEmpty()) {
+            line("options = pbandk.wkt.FieldOptions(").indented {
+                fieldOptions.deprecated?.let {
+                    lineBegin("deprecated = $it")
+                    if (fieldOptions.unknownFields.isEmpty()) lineEnd() else lineEnd(",")
+                }
+                if (fieldOptions.unknownFields.isNotEmpty()) {
+                    generateUnknownFields(fieldOptions.unknownFields)
+                }
+            }.line("),")
+        }
+    }
+
+    private fun generateUnknownFields(unknownFields: Map<Int, UnknownField>) {
+        line("unknownFields = mapOf(").indented {
+            val lastFieldIndex = unknownFields.size - 1
+            unknownFields.values.forEachIndexed { fieldIndex, field ->
+                line("${field.fieldNum} to pbandk.UnknownField(").indented {
+                    line("fieldNum = ${field.fieldNum},")
+                    val lastValueIndex = field.values.size - 1
+                    line("values = listOf(").indented {
+                        field.values.forEachIndexed { valueIndex, value ->
+                            lineBegin("pbandk.UnknownField.Value(")
+                            lineMid("wireType = ${value.wireType}, ")
+                            lineMid("rawBytes = byteArrayOf(${value.rawBytes.array.joinToString()})")
+                            lineMid(")")
+                            if (valueIndex != lastValueIndex) lineEnd(",") else lineEnd()
+                        }
+                    }.line(")")
+                }.lineBegin(")")
+                if (fieldIndex != lastFieldIndex) lineEnd(",") else lineEnd()
+            }
+        }.lineBegin(")")
```

Comment:
Ah, here we go. This needs to be `line` instead of `lineBegin`.

Reply:
fixed!